src/blockcipher/mod.rs

```diff
@@ -0,0 +1,198 @@
+// Copyright The ocicrypt Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Result};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::io::Read;
+
+pub mod aes_ctr;
+use aes_ctr::AESCTRBlockCipher;
+
+/// LayerCipherType is the ciphertype as specified in the layer metadata
+pub type LayerCipherType = String;
+
+/// TODO: Should be obtained from OCI spec once included
+pub const AES256CTR: &str = "AES_256_CTR_HMAC_SHA256";
+
+/// PrivateLayerBlockCipherOptions includes the information required to encrypt/decrypt
+/// an image layer which are sensitive and should not be in plaintext
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+pub struct PrivateLayerBlockCipherOptions {
+    /// symmetric_key represents the symmetric key used for encryption/decryption
+    /// This field should be populated by encrypt/decrypt calls
+    #[serde(rename = "symkey")]
+    pub symmetric_key: Vec<u8>,
+
+    /// digest is the digest of the original data for verification.
+    /// This is NOT populated by encrypt/decrypt calls
+    pub digest: String,
```

Comment:
Shouldn't we add the serde rename explicitly for all fields?

Reply:
Rust serialize and deserialize will use the parameter/rust name by default, we only need rename it when we have different name with the golang version.